packages/java/endpoint/src/main/java/com/vaadin/hilla/route/ClientRouteRegistry.java

```diff
@@ -84,11 +87,34 @@ public void removeRoute(String route) {
     public ClientViewConfig getRouteByPath(String path) {
         final Set<String> routes = registeredRoutes.keySet();
         final AntPathMatcher pathMatcher = new AntPathMatcher();
-        for (String route : routes) {
-            if (pathMatcher.match(route, path)) {
-                return registeredRoutes.get(route);
-            }
+        return Stream.of(addTrailingSlash(path), removeTrailingSlash(path))
+                .map(p -> {
+                    for (String route : routes) {
+                        if (pathMatcher.match(route, p)) {
+                            return registeredRoutes.get(route);
+                        }
+                    }
+                    return null;
+                }).filter(Objects::nonNull).findFirst().orElse(null);
+    }
```

Comment:
Just bypassing here: Did you consider security here? I remember some CVEs in the recent past which originated from security confusion based on trailing slash.. 

Additionally there is this recent Spring Framework Change to keep in mind: https://github.com/spring-projects/spring-framework/issues/28552

Reply:
This code doesn't actually change which content is rendered. Its purpose is to try to give the same view permission response that you would get on the client where the navigation is handled by the React Routed.

By doing some manual testing, I've seen that when two routes are present which are only different on the trailing slash, React Router always renders the one with the slash (e.g. you cannot see `/about` if `/about/` exists). But it is also forgiving: `/about/` will show the view for `/about` if an exact match is not there.

We don't have this information here, since the new File Router only stores the path items (e.g. `["about"]`). Also, the root path is stored as `""`, but it must match `"/"`.

This PR modifies the code that tries to find the route that React Router would render given a path. In that sense, it should not open any additional security breach.